### DISCUSSION TITLE
How to avoid screen reader read previous input

### DISCUSSION THREAD

**Discussion by ziwangj** [sourcelair/xterm.js#4561] [Q&A]
I use version 3.2 and enable the Screen Reader Mode. As far as I see, in this version, the input of the terminal is duplicated announced. Like:

Input1: (announced "w h  o a m i" and "whoami")
$whoami 

input2: (announced "d a t e" and "whoamidate")
$date

The whoami from previous input is not cleaned and announce again. This issue is never existed in latest version. But I do have the reason to use version 3.2. Could you help to clarify which element would impact screen reader to announce previous input and what's the potential fix for that? Thanks!

**Comment by jerch** [comment]
Version 3.2? Is this a typo? If you really mean 3.2 - well thats >5ys old with many newer releases in between (current is 5.2).

Idk if you can expect anyone to have an answer for your issue due to the very outdated version, you are prolly on your own here. 

The best advice I can give - you can still checkout the old code and start deep diving there. Once you have found a solution you can repackage the old version yourself containing the fix and use that bundle instead of the official one.

**Comment by ziwangj** [comment]
I would not expect someone to answer the issue but see if someone could help me to understand the mechanism of VoiceOver to read the terminal input. Because this issue is only represent for VoiceOver but not for JAWS and NVDA. Could you provide some direction about which attribute/class would it rely on to read terminal input? I did dive deep into the code. The only related difference  between this version and latest version. But after I did the change, it doesn't make any impact.

I tried to use version 5.1, but it also has some problem for VoiceOver in Mac.

**Reply by Tyriar** [reply]
We won't be able to help with old versions like that, but @meganrogge could help with the current state of voice over. We've had a push recently to improve voice over support in vscode.
